pkg/sqlutil/sqlutil.go

```diff
@@ -23,6 +24,23 @@ type DataSource interface {
 	NamedExecContext(ctx context.Context, query string, arg interface{}) (sql.Result, error)
 }
 
+// NamedQueryContext is like sqlx.NamedQueryContext, but it works with any DataSource.
```

Comment:
The sqlx version depends on concrete type switches instead of using an interface :facepalm: